### PR TITLE
Add support for IngressClass resources

### LINKED ISSUE #924: Support for k8s 1.18 Ingress Class?
How will nginx support the Ingress Class for the new k8s update?

```
apiVersion: "networking.k8s.io/v1beta1"
kind: "IngressClass"
metadata:
  name: "external-lb"
spec:
  controller: "example.com/ingress-controller" --> how should we specify nginx ingress controller?
  parameters: --> what parameters are needed?
    apiGroup: "k8s.example.com/v1alpha"
    kind: "IngressParameters"
    name: "external-lb"
```
Thank you!

### PR DESCRIPTION
**Pull request by lucacome**
Supporting `IngressClass` resource and `ingressClassName` for versions of k8s >= 1.18.

When using version of k8s >= 1.18 the `use-ingress-class-only` is now ignored and the Ingress Controller will only handle resources with the appropriate class set.

Fixes #924

### CODE REVIEW

**Review by LorcanMcVeigh — APPROVED** [review]

**Review by Dean-Coakley — COMMENTED** [review]
@lucacome This looks great!

Just one thing to make you aware of, is that you can set a compatible kubernetes version in your Chart.yaml. https://helm.sh/docs/topics/charts/#the-chartyaml-file

Example: `kubeVersion: 1.15-1.19`.

This would reject the installation of a helm release at the helm client level. I think this could be a nice improvement to include as it shortens the feedback loop for a user. (They don't have to install incompatible resources into their cluster to see logs that their cluster is incompatible.)
```
$ helm install test deployments/helm-chart
Error: chart requires kubeVersion: 1.15-1.19 which is incompatible with Kubernetes v1.14.0

$ helm ls
$
```

Feel free to ignore, we can always refer back to: https://docs.nginx.com/nginx-ingress-controller/technical-specifications/#supported-kubernetes-versions and hope most users read that.

File: docs-web/installation/running-multiple-ingress-controllers.md
Comment (pleshakov) [inline docs-web/installation/running-multiple-ingress-controllers.md:18]: You need to update the note in this section. Perhaps something like this? (please double check it, I might not have got it right with VirtualServers)

**Notes**: 
* For Kubernetes < 1.18,  if the class is not set in an Ingress configuration resource, the Ingress Controller will handle the resource. This is controlled via the `-use-ingress-class-only` argument. 
* For Kubernetes >= 1.18,  if the class is not set in an Ingress resource, Kubernetes will set it to the class of the default Ingress Controller. To make the Ingress Controller the default one, the `ingressclass.kubernetes.io/is-default-class` must be set on the IngressClass resource. See the Step 3 *Create an IngressClass resource* of the [Create Common Resources](https://deploy-preview-1133--nginx-kubernetes-ingress.netlify.app/nginx-ingress-controller/installation/installation-with-manifests/#create-common-resources) section.
* For VirtualServer and VirtualServerRoute resources, the Ingress Controller will always handle resources with an empty  class.

File: docs-web/installation/installation-with-helm.md
Comment (pleshakov) [inline docs-web/installation/installation-with-helm.md:277]: the preview for this table wasn't generated -- https://deploy-preview-1133--nginx-kubernetes-ingress.netlify.app/nginx-ingress-controller/installation/installation-with-helm/
I suspect some errors

File: deployments/helm-chart/Chart.yaml
Comment (pleshakov) [inline deployments/helm-chart/Chart.yaml:5]: In the main.go, we support 1.14. Here in helm, we don't support it. Was it intentional?  🤔

**Review by pleshakov — CHANGES_REQUESTED** [review]
Hi @lucacome 

Please see my comments and suggestions. Most of them about docs. Looks good overall.

Additionally,

(1) We're changing behavior for VirtualServer/VirtualServerRoutes, right? Previously, if `"-use-ingress-class-only" was enabled, the IC would ignore VS/VSRs with empty class. Now it won't ignore them. If that's the case, could you add the change label to the PR? We will also need to document that change in the changelog.

(2) Could you include in the description that you're fixing this https://github.com/nginxinc/kubernetes-ingress/issues/924 ?

**Review by Dean-Coakley — APPROVED** [review]
LGTM pending https://github.com/nginxinc/kubernetes-ingress/pull/1133#discussion_r485818176

(Sorry if my example misled you @lucacome 😓 )

File: docs-web/installation/running-multiple-ingress-controllers.md
Comment (pleshakov) [inline docs-web/installation/running-multiple-ingress-controllers.md:24]: same problem with an absolute link

**Review by pleshakov — APPROVED** [review]
@lucacome  looks good. Could you fix the mentioned link problems before merging?

**Pull request merged**
